src/NuGet.Core/NuGet.CommandLine.XPlat/Program.cs

```diff
@@ -55,6 +56,8 @@ public static int Main(string[] args)
 
             SetConnectionLimit();
 
+            SetUserAgent();
```

Comment:
nice catch. Can we make the UserAgent class throw if nothing have been set externally, so we force users to set it up? This should be a follow up issue